hail_search/hail_search_query.py

```diff
@@ -618,6 +630,62 @@ def _format_transcript_args(self):
         })
         return args
 
+    def _annotate_allowed_consequences(self, annotations, annotation_filters):
+        allowed_consequences = {
+            ann for field, anns in annotations.items()
+            if anns and (field not in ANNOTATION_OVERRIDE_FIELDS) for ann in anns
+        }
+        consequence_enum = self._get_enum_lookup('sorted_transcript_consequences', 'consequence_term')
+        allowed_consequence_ids = {consequence_enum[c] for c in allowed_consequences if consequence_enum.get(c)}
+
+        annotation_exprs = {}
+        if allowed_consequence_ids:
+            allowed_consequence_ids = hl.set(allowed_consequence_ids)
+            allowed_transcripts = self._ht.sorted_transcript_consequences.filter(
+                lambda tc: tc.consequence_term_ids.any(allowed_consequence_ids.contains)
+            )
+            annotation_exprs['allowed_transcripts'] = allowed_transcripts
+            annotation_filters = annotation_filters + [hl.is_defined(allowed_transcripts.first())]
+
+        if annotation_filters:
+            annotation_exprs['has_allowed_annotation'] = hl.any(annotation_filters)
+
+        if annotation_exprs:
+            self._ht = self._ht.annotate(**annotation_exprs)
+
+    def _get_annotation_override_filters(self, pathogenicity, annotations):
+        annotation_filters = []
+
+        for key, *args in self.PATHOGENICITY_FILTERS:
+            path_terms = (pathogenicity or {}).get(key)
+            if path_terms:
+                annotation_filters.append(self._has_terms_range_expr(path_terms, key, *args))
+        if annotations.get(SCREEN_KEY):
+            screen_enum = self._get_enum_lookup(SCREEN_KEY.lower(), 'region_type')
+            allowed_consequences = hl.set({screen_enum[c] for c in annotations[SCREEN_KEY]})
+            annotation_filters.append(allowed_consequences.contains(self._ht.screen.region_type_ids.first()))
+        if annotations.get(SPLICE_AI_FIELD):
+            score_filter, _ = self._get_in_silico_filter(SPLICE_AI_FIELD, annotations[SPLICE_AI_FIELD])
+            annotation_filters.append(score_filter)
+
+        return annotation_filters
+
+    def _has_terms_range_expr(self, terms, field, subfield, range_configs):
+        enum_lookup = self._get_enum_lookup(field, subfield)
+
+        ranges = [[None, None]]
+        for path_filter, start, end in range_configs:
+            if path_filter in terms:
+                ranges[-1][1] = len(enum_lookup) if end is None else enum_lookup[end]
```

Comment:
This is maybe out of the scope of this pr, but would it make sense to do the enum lookups of the statically configured ranges at startup?

Reply:
The lookup is pretty fast and we need to evaluate the enum for formatting results anyways, so it doesn't really add any time relative to the code before this PR. Theres maybe a bigger question of should we read all the enums and globals at application startup instead of on a per search basis, which might be faster in the long run. But I think thats out of scope of this PR. If you think thats a better approach, I can make a follow up for that

Reply:
Yeah, I think doing the enums and globals at application startup would make more sense than at query time.  And yeah, follow up make more sense than on this pr.